src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -163,8 +164,10 @@ private static ImmutableArray<Diagnostic> EmitRegexMethod(IndentedTextWriter wri
             writer.Write("    public static global::System.Text.RegularExpressions.Regex Instance { get; } = ");
 
             // If we can't support custom generation for this regex, spit out a Regex constructor call.
-            if (!SupportsCodeGeneration(rm))
+            if (!SupportsCodeGeneration(rm, out string? reason))
             {
+                writer.WriteLine();
+                writer.WriteLine($"// Cannot generate optimized code because {reason}.");
```

Comment:
```suggestion
                writer.WriteLine($"// Cannot generate Regex-derived implementation because {reason}.");
```

Reply:
I liked "optimized code" because it is more approachable, but could imply that the generic `Regex` object is not optimized at all.